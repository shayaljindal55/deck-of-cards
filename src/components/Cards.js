import React from 'react'
import '../styles/style.css';

export default function Cards(props) {
    const { suit, cardValue, color, id } = props;
    return (
        <div className={`outline outline-adjusted shadow rounded ${color}`}
            key={id}
            data-testid={id}>
            <div className="top">
                <span data-testid={`cardValue ${id}`}>{cardValue}</span>
                <span data-testid={`suit ${id}`}>{suit}</span>
            </div>
        </div>
    )
}
