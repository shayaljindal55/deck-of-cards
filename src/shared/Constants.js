
export const suits = ["♣︎", "♠︎", "♥︎", "♦︎"]; // defaulted to required sort order
export const values = ["2", "3", "4", "5", "6", "7", "8", "9", "10", "J", "Q", "K", "A"];
export const redSuits = ["♥︎", "♦︎"]
export const noDrawnCardSort = "There are no drawn cards to sort. Please draw cards from the deck."
export const noCardsInDeck = "No cards in the deck."
export const testSampleDrawnCards = `[
{"id":40,"cardValue":"3","suit":"♦︎","color":"red","suitIndex":3,"rankIndex":1},
{"id":35,"cardValue":"A","suit":"♥︎","color":"red","suitIndex":2,"rankIndex":9},
{"id":34,"cardValue":"10","suit":"♥︎","color":"red","suitIndex":2,"rankIndex":8},
{"id":13,"cardValue":"2","suit":"♠︎","color":"black","suitIndex":1,"rankIndex":0},
{"id":7,"cardValue":"9","suit":"♣︎","color":"black","suitIndex":0,"rankIndex":7}]`
export const testExpectedSortedDrawnCards = `[
    {"id":7,"cardValue":"9","suit":"♣︎","color":"black","suitIndex":0,"rankIndex":7},
    {"id":13,"cardValue":"2","suit":"♠︎","color":"black","suitIndex":1,"rankIndex":0},
    {"id":34,"cardValue":"10","suit":"♥︎","color":"red","suitIndex":2,"rankIndex":8},
    {"id":35,"cardValue":"A","suit":"♥︎","color":"red","suitIndex":2,"rankIndex":9},
    {"id":40,"cardValue":"3","suit":"♦︎","color":"red","suitIndex":3,"rankIndex":1}
   ]`
