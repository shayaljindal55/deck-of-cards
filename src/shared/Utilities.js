export const sortCards = (drawnCards) => {
    return drawnCards.sort(function (a, b) {
        if (a.suitIndex < b.suitIndex ||
            (a.suitIndex === b.suitIndex && a.rankIndex < b.rankIndex)) {
            return -1;
        }
        return 1;
    });
};