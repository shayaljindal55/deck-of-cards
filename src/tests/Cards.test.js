import Cards from '../components/Cards';
import { suits, values } from '../shared/Constants';
import { render, screen } from '@testing-library/react';
/* react-test-renderer is a library for rendering React components to pure JavaScript objects */
import { create } from "react-test-renderer";

const props = {
    suit: suits[0],
    value: values[0],
    color: "black",
    id: "1"
};
describe("Cards component render with valid props validation", () => {
    test('verify DOM with expected prop values', () => {
        render(CardsComponent());
        /* test that all elements with correct prop values are present in DOM */
        verifyElementInDOM();
    });
});
function verifyElementInDOM(id) {
    [props.id, `cardValue ${props.id}`, `suit ${props.id}`].forEach(id => {
        const ele = screen.queryByTestId(id);
        expect(ele).toBeInTheDocument();
    });
}

describe("Cards component", () => {
    test("Matches the snapshot", () => {
        /*  create is a method from react-test-renderer for "mounting" the component */
        const card = create(<CardsComponent />);
        expect(card.toJSON()).toMatchSnapshot();
    });
});

function CardsComponent() {
    return <Cards suit={props.suit}
        cardValue={props.value}
        color={props.color} id={props.id} />;
}