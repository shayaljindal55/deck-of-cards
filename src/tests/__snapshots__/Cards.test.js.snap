// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Cards component Matches the snapshot 1`] = `
<div
  className="outline outline-adjusted shadow rounded black"
  data-testid="1"
>
  <div
    className="top"
  >
    <span
      data-testid="cardValue 1"
    >
      2
    </span>
    <span
      data-testid="suit 1"
    >
      ♣︎
    </span>
  </div>
</div>
`;
